File: Examples/VisualBasic/Loading-and-Saving/OpenSceneFromProtectedPdf.vb
Imports System.IO
Imports System.Text
Imports System.Collections
Imports Aspose.ThreeD
Imports Aspose.ThreeD.Formats

Namespace Loading_Saving
    Class OpenSceneFromProtectedPdf
        Public Shared Sub Run()
            ' ExStart:OpenSceneFromProtectedPdf
            ' The path to the documents directory.
            Dim MyDir As String = RunExamples.GetDataDir()
            ' Create a new scene
            Dim scene As New Scene()
            ' Use loading options and apply password
            Dim opt As New PdfLoadOptions() With {
                 .Password = Encoding.UTF8.GetBytes("password") _
            }
            ' Open scene
            scene.Open(MyDir & Convert.ToString("House_Design.pdf"), opt)
            ' ExEnd:OpenSceneFromProtectedPdf            
        End Sub
    End Class
End Namespace

File: Examples/CSharp/Loading-and-Saving/OpenSceneFromProtectedPdf.cs
using System;
using System.IO;
using System.Text;
using System.Collections;
using Aspose.ThreeD;
using Aspose.ThreeD.Formats;

namespace Aspose._3D.Examples.CSharp.Loading_Saving
{
    class OpenSceneFromProtectedPdf
    {
        public static void Run()
        {
            // ExStart:OpenSceneFromProtectedPdf
            // Create a new scene
            Scene scene = new Scene();
            // Use loading options and apply password
            PdfLoadOptions opt = new PdfLoadOptions() { Password = Encoding.UTF8.GetBytes("password") };
            // Open scene
            scene.Open(RunExamples.GetDataFilePath("House_Design.pdf"), opt);
            // ExEnd:OpenSceneFromProtectedPdf            
        }
    }
}
